tooling/lsp/src/requests/inlay_hint.rs

```diff
@@ -584,64 +838,92 @@ mod inlay_hints_tests {
         assert!(!inlay_hints.is_empty());
     }
-}
 
-// These functions are copied from the codespan_lsp crate, except that they never panic
-// (the library will sometimes panic, so functions returning Result are not always accurate)
+    #[test]
+    async fn test_dont_collect_parameter_inlay_hints_if_disabled() {
+        let inlay_hints = get_inlay_hints(24, 26, no_hints()).await;
+        assert!(inlay_hints.is_empty());
+    }
 
-fn range_to_byte_span(
-    files: &FileMap,
-    file_id: FileId,
-    range: &lsp_types::Range,
-) -> Option<std::ops::Range<usize>> {
-    Some(
-        position_to_byte_index(files, file_id, &range.start)?
-            ..position_to_byte_index(files, file_id, &range.end)?,
-    )
-}
+    #[test]
+    async fn test_collect_parameter_inlay_hints_in_function_call() {
+        let inlay_hints = get_inlay_hints(24, 26, parameter_hints()).await;
+        assert_eq!(inlay_hints.len(), 2);
 
-fn position_to_byte_index(
-    files: &FileMap,
-    file_id: FileId,
-    position: &lsp_types::Position,
-) -> Option<usize> {
-    let Ok(source) = files.source(file_id) else {
-        return None;
-    };
+        let inlay_hint = &inlay_hints[0];
+        assert_eq!(inlay_hint.position, Position { line: 25, character: 12 });
+        if let InlayHintLabel::String(label) = &inlay_hint.label {
+            assert_eq!(label, "one: ");
+        } else {
+            panic!("Expected InlayHintLabel::String, got {:?}", inlay_hint.label);
+        }
 
-    let Ok(line_span) = files.line_range(file_id, position.line as usize) else {
-        return None;
-    };
-    let line_str = source.get(line_span.clone())?;
+        let inlay_hint = &inlay_hints[1];
+        assert_eq!(inlay_hint.position, Position { line: 25, character: 15 });
+        if let InlayHintLabel::String(label) = &inlay_hint.label {
+            assert_eq!(label, "two: ");
+        } else {
+            panic!("Expected InlayHintLabel::String, got {:?}", inlay_hint.label);
+        }
+    }
 
-    let byte_offset = character_to_line_offset(line_str, position.character)?;
+    #[test]
+    async fn test_collect_parameter_inlay_hints_in_method_call() {
+        let inlay_hints = get_inlay_hints(36, 39, parameter_hints()).await;
+        assert_eq!(inlay_hints.len(), 1);
 
-    Some(line_span.start + byte_offset)
-}
+        let inlay_hint = &inlay_hints[0];
+        assert_eq!(inlay_hint.position, Position { line: 38, character: 18 });
+        if let InlayHintLabel::String(label) = &inlay_hint.label {
+            assert_eq!(label, "one: ");
+        } else {
+            panic!("Expected InlayHintLabel::String, got {:?}", inlay_hint.label);
+        }
+    }
 
-fn character_to_line_offset(line: &str, character: u32) -> Option<usize> {
-    let line_len = line.len();
-    let mut character_offset = 0;
+    #[test]
+    async fn test_dont_show_parameter_inlay_hints_if_name_matches_var_name() {
```

Comment:
Please add `dont` to the cspell file. (I'm a bit surprised it's not already supported)

Reply:
Could also just do `do_not`